dep-graph/client/src/app/sidebar/search-depth.tsx

```diff
@@ -0,0 +1,98 @@
+export interface SearchDepthProps {
+  searchDepth: number;
+  searchDepthEnabled: boolean;
+  searchDepthFilterEnabledChange: (checked: boolean) => void;
+  decrementDepthFilter: () => void;
+  incrementDepthFilter: () => void;
+}
+
+export function SearchDepth({
```

Comment:
Consider wrapping this with `React.memo` since this component only needs to render when props change.